slackapi/slackbot-bolt.mjs

```diff
@@ -0,0 +1,312 @@
+slackApp.command('/addbook', async ({ ack, body, client }) => {
+  await ack();
+
+  try {
+    const result = await client.views.open({
+      trigger_id: body.trigger_id,
+      view: {
+        "title": {
+          "type": "plain_text",
+          "text": "Add a book"
+        },
+        "submit": {
+          "type": "plain_text",
+          "text": "Submit"
+        },
+        "blocks": [
+          {
+            "type": "input",
+            "block_id": "input_book_name",
+            "element": {
+              "type": "plain_text_input",
+              "action_id": "input_name",
+              "placeholder": {
+                "type": "plain_text",
+                "text": "Book name"
+              }
+            },
+            "label": {
+              "type": "plain_text",
+              "text": "Title"
+            }
+          },
+          {
+            "type": "input",
+            "block_id": "input_book_author",
+            "element": {
+              "type": "plain_text_input",
+              "action_id": "input_author",
+              "placeholder": {
+                "type": "plain_text",
+                "text": "Author name"
+              }
+            },
+            "label": {
+              "type": "plain_text",
+              "text": "Author",
+              "emoji": true
+            }
+          },
+          {
+            "type": "input",
+            "block_id": "input_book_abstract",
+            "element": {
+              "type": "plain_text_input",
+              "action_id": "input_abstract",
+              "multiline": true
+            },
+            "label": {
+              "type": "plain_text",
+              "text": "Abstract"
+            },
+            "hint": {
+              "type": "plain_text",
+              "text": "Description of the book"
+            }
+          },
+          {
+            "type": "input",
+            "block_id": "input_book_publication",
+            "element": {
+              "type": "datepicker",
+              "initial_date": "1990-04-28",
+              "placeholder": {
+                "type": "plain_text",
+                "text": "Select the publication date",
+                "emoji": true
+              },
+              "action_id": "input_publication"
+            },
+            "label": {
+              "type": "plain_text",
+              "text": "Publication date",
+              "emoji": true
+            }
+          },
+          {
+            "type": "input",
+            "block_id": "input_book_cover",
+            "element": {
+              "type": "plain_text_input",
+              "action_id": "input_cover",
+              "placeholder": {
+                "type": "plain_text",
+                "text": "https://url.image"
+              }
+            },
+            "label": {
+              "type": "plain_text",
+              "text": "Cover URL",
+              "emoji": true
+            },
+            "optional": true
+          }
```

Comment:
Maybe you can extract this json from an external file and just import it

Reply:
Sure I do :D thanks